dts/bindings/kscan/zephyr,kscan-input.yaml

```diff
@@ -0,0 +1,23 @@
+# Copyright 2023 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Input to kscan adapter.
+
+  Allows using an input device with the kscan API. Define as a child node of
+  the input device, for example
+
+  chosen {
+    zephyr,keyboard-scan = &kscan_input;
+  };
+
+  ft5336@38 {
+    ...
+    kscan_input: kscan-input {
+      compatible = "zephyr,kscan-input";
+    };
```

Comment:
Just a comment here: we're creating this node because of how Zephyr creates devices (1 device = 1 dt node). We can't have an input+kscan device referring to the same DT node. Considering this is a hard limitation of our existing device model, I understand there's no better way to do this.

From https://docs.kernel.org/devicetree/bindings/writing-bindings.html
> DON’T create nodes just for the sake of instantiating drivers. Multi-function devices only need child nodes when the child nodes have their own DT resources. A single node can be multiple providers (e.g. clocks and resets).

We should fix this imposed limitation, as this effectively forces us to deviate from DT _good practices_.

Reply:
Yeah good point, but sharing a device pointer would not work here anyway if the input device was implementing a subsystem API (which I suspect will happen eventually).

Reply:
I actually have a prototype in a branch where you could have N devices = 1 dt node, provided each device has a different class. It's of course a breaking change to the device model, but I think it's worth exploring. The current model isn't great, has many obvious limitations and has never evolved from just being a "pointer container". I'll try to retake that work soon (busy now with some other stuff).